unstructured/partition/auto.py

```diff
@@ -471,10 +471,11 @@ def file_and_type_from_url(
     response = requests.get(url, headers=headers, verify=ssl_verify, timeout=request_timeout)
     file = io.BytesIO(response.content)
 
-    content_type = (
-        content_type or response.headers.get("Content-Type", "").split(";")[0].strip().lower()
-    )
-    encoding = response.headers.get("Content-Encoding", "utf-8")
+    if content_type := content_type or response.headers.get("Content-Type", None):
+        content_type = content_type.split(";")[0].strip().lower()
+
+    # -- non-None when response is textual --
+    encoding = response.encoding
```

Comment:
This was just plain wrong before. Not sure exactly why the test used to pass. The `Content-Encoding:` header describes the compression scheme used for downloading the HTTP response (`"gzip" in this case) so was not the right thing at all. `encoding` is how to decode the bytes of a textual file-format into a `str`, so like "utf-8" or similar.